### PR TITLE
update(platform): BYOC GCP LA MS1

### PR DESCRIPTION
**Pull request by wojcik-dorota**
https://aiven.atlassian.net/browse/MILK-7

- GCP - custom cloud creation via CLI only
- Private and public deployment models for AWS and GCP
- Tagging
  - Infra tags via CLI
  - Service tags via console
- Permissions

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

### CODE REVIEW

File: static/images/content/figma/byoc-gcp-private.png
Comment (Prime541) [inline static/images/content/figma/byoc-gcp-private.png:1]: Now there is only 1 BYOC VPC for Google private model.

File: docs/platform/howto/byoc/create-custom-cloud.md [hunk 1]
```diff
@@ -657,11 +704,97 @@ cloud** view.
...
+    Before adding ingress rules, see the examples provided in the
+    Terraform template you generated and downloaded from [Aiven
+    Console](https://console.aiven.io/).
+    :::
+
+1. Use Terraform to deploy the infrastructure template with the provided variables in your
+   GCP account. This will generate a privilege-bearing service account (SA).
+
+    :::important
+    When running `terraform plan` and `terraform apply`, make sure you
```
Comment (Prime541) [inline docs/platform/howto/byoc/create-custom-cloud.md:756]: `terraform destroy` needs this option too.

File: docs/platform/howto/byoc/create-custom-cloud.md [hunk 2]
```diff
@@ -713,13 +713,67 @@ Your new custom cloud is ready to use only after its status changes to
...
+    </summary>
+
+    ```json
+    {
+        "custom_cloud_environment": {
+            "cloud_provider": "google",
+            "cloud_region": "europe-north1",
+            "contact_emails": [
+                {
+                    "email": "heikju@aiven.fi",
```
Comment (Prime541) [inline docs/platform/howto/byoc/create-custom-cloud.md:738]: I wonder if we should anonymize some data for the public doc?

File: docs/platform/howto/byoc/create-custom-cloud.md [hunk 3]
```diff
@@ -713,13 +713,67 @@ Your new custom cloud is ready to use only after its status changes to
 <TabItem value="2" label="GCP">
 
 1. Generate an IaC template by running [avn byoc create](/docs/tools/cli/byoc#avn-byoc-create).
+
+   ```bash
+   avn byoc create                               /
```
Comment (Prime541) [inline docs/platform/howto/byoc/create-custom-cloud.md:718]: The continuation of a line in shell is `\` backslash at the end of the line, not `/` slash.

File: docs/platform/howto/byoc/create-custom-cloud.md [hunk 4]
```diff
@@ -1,22 +1,23 @@
 ---
-title: Create an AWS custom cloud in Aiven
+title: Create a custom cloud
 sidebar_label: Create custom clouds
-keywords: [bring your own cloud, byoc, custom cloud]
+keywords: [AWS, Amazon Web Services, GCP, Google Cloud Platform, byoc, bring your own cloud, custom cloud]
```
Comment (Prime541) [inline docs/platform/howto/byoc/create-custom-cloud.md:4]: What about Azure and OCI?

File: docs/platform/howto/byoc/create-custom-cloud.md [hunk 5]
```diff
@@ -65,33 +87,51 @@ contacts for your custom cloud.
...
+
+<Tabs groupId="group1">
+<TabItem value="1" label="AWS permissions" default>
 <details><summary>
-Show permissions required for creating resources for bastion and
-workload networks
+Show permissions required for creating resources for bastion and workload networks
 </summary>
 
 ```bash
```
Comment (Prime541) [inline docs/platform/howto/byoc/create-custom-cloud.md:137]: This is not `bash` but `json`.

File: docs/platform/howto/byoc/create-custom-cloud.md [hunk 6]
```diff
@@ -440,47 +480,48 @@ workload networks
...
+
+- `get` on each region to retrieve region-level quotas
+
+- `ListTimeSeriesRequest` using the Google `monitoring_v3.MetricServiceClient` for the
+  following metrics:
+
+  - `compute.googleapis.com/quota/static_routes_per_vpc_network/usage`
+  - `compute.googleapis.com/quota/static_routes_per_vpc_network/limit`
+
+</details>
```
Comment (Prime541) [inline docs/platform/howto/byoc/create-custom-cloud.md:500]: I wonder where these details come from?

I would say that the customer service account that will run the TF script in the Customer Google project needs the permissions listed in the following Google [predefined roles](https://cloud.google.com/iam/docs/understanding-roles):
- roles/iam.serviceAccountAdmin (setup impersonation to Privilege-Bearing Service Account)
- roles/resourcemanager.projectIamAdmin (grant use of Customer project to P-B SA)
- roles/compute.instanceAdmin.v1 (manage networks and instances)
- roles/compute.securityAdmin (create firewall rules)
- enable [IAM API](https://console.developers.google.com/apis/api/iam.googleapis.com) to create P-B SA
- enable [CRM API](https://console.cloud.google.com/apis/library/cloudresourcemanager.googleapis.com) to set IAM policies to P-B SA.

File: docs/platform/howto/byoc/create-custom-cloud.md [hunk 7]
```diff
@@ -572,8 +624,8 @@ AWS account. Continue working in the **Create custom cloud** wizard:
     provided variables.
 
     :::important
-    When running `terraform plan` and `terraform apply`, make sure you
-    add `-var-file=FILE_NAME.vars` as an option.
+    When running `terraform plan` and `terraform apply`, add `-var-file=FILE_NAME.vars`
```
Comment (Prime541) [inline docs/platform/howto/byoc/create-custom-cloud.md:627]: and `terraform destroy`.

File: docs/platform/howto/byoc/download-infrastructure-template.md [hunk 8]
```diff
@@ -5,29 +5,61 @@ keywords: [Terraform, deployment, deploy, byoc, bring your own cloud, custom clo
...
 1.  Click <ConsoleLabel name="download"/>.
 
-## Download the template via CLI
+#### Via the Aiven CLI client
 
 [Run the `avn byoc template terraform get-template` command](/docs/tools/cli/byoc#avn-byoc-template-terraform-get-template)
 to download your infrastructure template.
+</TabItem>
+<TabItem value="2" label="GCP">
+[Run the `avn byoc template terraform get-template` command](/docs/tools/cli/byoc#avn-byoc-template-terraform-get-template)
```
Comment (Prime541) [inline docs/platform/howto/byoc/download-infrastructure-template.md:56]: Should we tell about getting the variable files too?

File: docs/platform/howto/byoc/tag-custom-cloud-resources.md [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+import ConsoleLabel from "@site/src/components/ConsoleIcons"
+
+Custom cloud tags propagate to resources on the Aiven platform and in your own cloud infrastructure. Tagging allows resource categorization, which simplifies governance, cost allocation, and system performance review.
+
+:::important
+Since the tags propagate to your own cloud infrastructure, the contents and total number
+of the tags need to stay within limits imposed by your cloud provider.
+:::
+
+## Types of tagging
```
Comment (Prime541) [inline docs/platform/howto/byoc/tag-custom-cloud-resources.md:17]: Beware that for tagging GCP BYOC we use Google labels, not Google tags.

File: docs/platform/concepts/byoc.md [hunk 10]
```diff
@@ -47,15 +52,15 @@ needs or project requirements, such as:
 ## Who is eligible for BYOC {#eligible-for-byoc}
 
 The BYOC setup is a bespoke service offered on a case-by-case basis, and
-not all cloud providers support it yet. Meet a few requirements to be eligible for BYOC:
+not all cloud providers support it yet. You're eligible for BYOC if:
 
--   You use Amazon Web Services (AWS), Google Cloud, or Microsoft Azure (excluding
-    Azure Germany).
+-   You use Amazon Web Services (AWS), Google Cloud Platform (GCP), Microsoft Azure
+    (excluding Azure Germany), or Oracle Cloud Infrastructure (OCI).
```
[the remark this answers is not in the document]
Comment (wojcik-dorota) [reply docs/platform/concepts/byoc.md:58]: Yes, this specific article should cover the BYOC feature we support both via self-service (AWS, GCP) and via SRE (Azure, OCI). The differentiation comes in how-to docs.

File: docs/platform/howto/byoc/create-custom-cloud.md (hunk 7 above)
Comment (wojcik-dorota) [reply docs/platform/howto/byoc/create-custom-cloud.md:627]: As pointed out by @jeremy-heller-aiven , we should not use `terraform destroy` in the Create a custom cloud` flow hence I'm not including it. cc @jeremy-heller-aiven

**Review by Prime541 — APPROVED** [review]
LGTM, just a few not important things in remarks

File: docs/platform/howto/byoc/assign-project-custom-cloud.md [hunk 11]
```diff
@@ -1,111 +1,108 @@
...
 
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
 import ConsoleLabel from "@site/src/components/ConsoleIcons";
 
-Configure the availability of your [custom cloud](/docs/platform/concepts/byoc) to access
-it in all the projects, selected organizational units, or specific projects only.
+Select your organizations, units, or project that can access and use your [custom cloud](/docs/platform/concepts/byoc).
 
 ## About making custom clouds available from your projects
```
Comment (jeremy-heller-aiven) [inline docs/platform/howto/byoc/assign-project-custom-cloud.md:12]: Maybe change "About making" to "How to make"

File: docs/platform/howto/byoc/assign-project-custom-cloud.md [hunk 12]
```diff
@@ -1,111 +1,108 @@
...
+by following
+[Enable projects to use your custom cloud](#enable-projects-to-use-your-custom-cloud).
+
+You can make your custom cloud available by default in all existing and future projects,
+or you can pick specific projects and/or organizational units where you want your custom
+cloud to be available.
 
 ## Prerequisites
 
--   Administrator's role for your Aiven organization
```
Comment (jeremy-heller-aiven) [inline docs/platform/howto/byoc/assign-project-custom-cloud.md:37]: Is Super Admin not required to assign projects?

**Review by jeremy-heller-aiven — CHANGES_REQUESTED** [review]

File: docs/platform/howto/byoc/assign-project-custom-cloud.md (hunk 11 above)
Comment (wojcik-dorota) [reply docs/platform/howto/byoc/assign-project-custom-cloud.md:12]: Actually, this section is only to give some context for the upcoming operation the user will to introduced to in the next sections. Saying "How to" might set some expectations for steps required for enabling projects on custom clouds, which is not the purpose of this "About.." section

**Review by ArthurFlag — CHANGES_REQUESTED** [review]
Minor comments, looks good otherwise 🙌

File: sidebars.ts [hunk 13]
```diff
@@ -281,6 +281,7 @@ const sidebars: SidebarsConfig = {
                 'platform/howto/byoc/create-custom-cloud',
                 'platform/howto/byoc/assign-project-custom-cloud',
                 'platform/howto/byoc/add-customer-info-custom-cloud',
+                'platform/howto/byoc/tag-custom-cloud-resources',
```
Comment (ArthurFlag) [inline sidebars.ts:284]: Should we have `rename-custom-cloud` and  `delete-custom-cloud` next to one another?

File: docs/platform/howto/byoc/rename-custom-cloud.md [hunk 14]
```diff
@@ -1,63 +1,75 @@
...
 sidebar_label: Rename custom clouds
 ---
 
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
 import ConsoleLabel from "@site/src/components/ConsoleIcons";
 
 Change the name of your [custom cloud](/docs/platform/concepts/byoc).
 
 ## About renaming custom clouds
```
Comment (ArthurFlag) [inline docs/platform/howto/byoc/rename-custom-cloud.md:12]: Same comment about the "about titles".

**Review by ArthurFlag — APPROVED** [review]

**Review by jeremy-heller-aiven — APPROVED** [review]

**Comment by Prime541** [comment]
Who should merge this?

**Comment by wojcik-dorota** [comment]
> Who should merge this?

I'll do that as soon as we officially release and @jeremy-heller-aiven gives the green light.

**Pull request merged**